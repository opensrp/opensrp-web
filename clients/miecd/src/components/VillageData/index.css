#navrow {
  display: flex;
  justify-content: center;
}

.village .card {
  width: 100%;
}

.village td:first-of-type,
.village th:first-of-type {
  width: 9%;
}

.village .card-body {
  padding: 0;
}

.village .table-card {
  background-color: #ffffff;
  border: 1px solid #dadce0;
  border-radius: 5px;
  box-shadow: 0 1px 3px 0 #d7d7e0;
  width: 100%;
  margin: 0 !important;
}

.village .table-card thead > tr {
  border: 1px solid #dadce0;
}

.village .card-title {
  background-color: #ffffff;
  border: 1px solid #dadce0;
  color: #3a484f;
  font-size: 1rem;
  font-weight: 500;
  letter-spacing: -0.018rem;
  line-height: 1rem;
  text-align: left;
  padding: 15px 20px;
}

.village .rt-tr.-odd {
  background-color: #f8f9fa;
}

.back-page > span > span {
  padding-left: 15px;
}

.village .ReactTable .rt-thead.-header {
  box-shadow: 0 0px 0px 0 #dadce0;
}

.village table td {
  border: 1px solid #dadce0;
  text-align: center;
}

.village a,
.village a:hover,
.village a:active,
.village a:visited {
  color: #0292d3;
  text-decoration: none;
}

.village table td:nth-child(1) {
  text-align: left;
  padding-left: 1%;
  color: #0292d3;
  font-size: 15px;
  font-weight: 400;
  line-height: 18px;
}

.village .card-title > span {
  color: #7c7c7c;
}

.village .rt-th.rt-resizable-header.-cursor-pointer,
.village table td {
  color: #3a484f;
  font-size: 14px;
  font-weight: 400;
  line-height: 17px;
  width: 57px;
}

.village .table-container table #body {
  overflow: auto;
}

.village > h1 {
  padding: 15px 0;
}

.village #header {
  position: relative;
  top: 0;
  height: 50px;
  border-bottom: 0.5px solid #dadce0;
}

.village .default-width {
  width: 20%;
  padding: 20px;
}

.village table {
  margin: 0;
}

.village table thead tr th {
  text-align: center;
  color: #3a484f;
  font-size: 15px;
  font-weight: 400;
}

.village table tbody tr:first-child td {
  border-top: none;
}

.village td#total,
.village #header {
  color: #3a484f;
  font-size: 0.875rem;
  font-weight: 400;
  line-height: 1.063rem;
}

.village table.table.table-borderless.table-striped,
.village .card-title {
  margin: 0;
  border: none;
}

.village td.default-width {
  border: 1px solid #dadce0;
  margin: 0;
}

.village #no-rows > td {
  width: 100%;
}

.village .high {
  background-color: #ffe8e6;
}

.village .low {
  background-color: #ffe9db;
}

.village .no {
  background-color: rgba(55, 174, 18, 0.10000000149011612);
}

span.divider {
  color: #0292d3 !important;
}
