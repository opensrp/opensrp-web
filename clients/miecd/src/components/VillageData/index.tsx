import React from 'react';
import { Link } from 'react-router-dom';
import { Card, CardBody, CardTitle, Row, Table } from 'reactstrap';
import {
    CURRENT_SYMPTOMS,
    CURRENT_WEIGHT,
    DAYS_SINCE_BIRTH,
    EDD,
    GRAVIDITY,
    LOCATION,
    LOCATION_OF_BIRTH,
    LOCATION_OF_RESIDENCE,
    NBC_AND_PNC_CHILD,
    NUTRITION,
    PARITY,
    PATIENT_ID,
    PREGNANCY,
    PREVIOUS_PREGNANCY_RISK,
    RISK_CATEGORY,
} from '../../constants';
import { getModuleLink, getNumberOfDaysSinceDate } from '../../helpers/utils';
import { SmsData } from '../../store/ducks/sms_events';
import { PaginationData, Paginator, PaginatorProps } from '../Paginator';
import RiskColoring from '../RiskColoring';
import './index.css';

export interface Props {
    current_level: number;
    smsData: SmsData[];
    module: string;
}

interface State {
    currentPage: number;
}

const defaultProps: Props = {
    // eslint-disable-next-line @typescript-eslint/camelcase
    current_level: 0,
    module: '',
    smsData: [],
};

export default class VillageData extends React.Component<Props, State> {
    public static defaultProps = defaultProps;

    constructor(props: Props) {
        super(props);
        this.state = {
            currentPage: 1,
        };
    }

    public render() {
        const routePaginatorProps: PaginatorProps = {
            endLabel: 'last',
            nextLabel: 'next',
            onPageChange: (paginationData: PaginationData) => {
                this.setState({
                    currentPage: paginationData.currentPage,
                });
            },
            pageLimit: 5,
            pageNeighbours: 3,
            previousLabel: 'previous',
            startLabel: 'first',
            totalRecords: this.props.smsData.length,
        };

        return (
            <React.Fragment>
                {this.props.current_level >= 3 ? (
                    <React.Fragment>
                        <Row className="village villageDataRow">
                            <Card className="table-card">
                                <CardTitle>Selected Commune</CardTitle>
                                <CardBody>
                                    <Table striped={true} borderless={true}>
                                        <thead id="header">
                                            {this.props.module === PREGNANCY ? (
                                                <tr>
                                                    <th className="default-width">{PATIENT_ID}</th>
                                                    <th className="default-width">{GRAVIDITY}</th>
                                                    <th className="default-width">{PARITY}</th>
                                                    <th className="default-width">{LOCATION}</th>
                                                    <th className="default-width">{EDD}</th>
                                                    <th className="default-width">{PREVIOUS_PREGNANCY_RISK}</th>
                                                    <th className="default-width">{RISK_CATEGORY}</th>
                                                </tr>
                                            ) : this.props.module === NUTRITION ? (
                                                <tr>
                                                    <th className="default-width">{PATIENT_ID}</th>
                                                    <th className="default-width">{DAYS_SINCE_BIRTH}</th>
                                                    <th className="default-width">{LOCATION_OF_RESIDENCE}</th>
                                                    <th className="default-width">{CURRENT_WEIGHT}</th>
                                                    <th className="default-width">{LOCATION_OF_BIRTH}</th>
                                                    <th className="default-width">{RISK_CATEGORY}</th>
                                                </tr>
                                            ) : (
                                                <tr>
                                                    <th className="default-width">{PATIENT_ID}</th>
                                                    <th className="default-width">{DAYS_SINCE_BIRTH}</th>
                                                    <th className="default-width">{LOCATION_OF_RESIDENCE}</th>
                                                    <th className="default-width">{CURRENT_SYMPTOMS}</th>
                                                    <th className="default-width">{LOCATION_OF_BIRTH}</th>
                                                    <th className="default-width">{RISK_CATEGORY}</th>
                                                </tr>
                                            )}
                                        </thead>
                                        <tbody id="body">
                                            {this.props.smsData.length
                                                ? this.props.smsData
                                                      .slice(
                                                          (this.state.currentPage - 1) * routePaginatorProps.pageLimit,
                                                          (this.state.currentPage - 1) * routePaginatorProps.pageLimit +
                                                              routePaginatorProps.pageLimit,
                                                      )
                                                      .map(
                                                          this.props.module === PREGNANCY
                                                              ? this.pregnancyMapFunction
                                                              : this.props.module === NBC_AND_PNC_CHILD
                                                              ? this.nbcAndPncChildMapFunction
                                                              : this.props.module === NUTRITION
                                                              ? this.nutritionMapFunction
                                                              : this.nbcAndPncMotherMapFunction,
                                                      )
                                                : null}
                                        </tbody>
                                    </Table>
                                </CardBody>
                            </Card>
                        </Row>
                        <Row id="navrow" className="villageDataRow">
                            <Paginator {...routePaginatorProps} />
                        </Row>
                    </React.Fragment>
                ) : null}
            </React.Fragment>
        );
    }

    /**
     * Returns a <tr></tr> for an SmsData object passed to it
     * which is used to build the table above.
     * for the NBC & PNC_MOTHER module
     * @param {SmsData} dataItem - an SmsData object used to generate a table row
     * @return {JSX.Element} table row
     */
    public nbcAndPncMotherMapFunction = (dataItem: SmsData): JSX.Element => {
        return (
            <tr key={dataItem.event_id}>
                <td className="default-width">
                    <Link to={`${getModuleLink(this.props.module)}/patient_detail/${dataItem.anc_id}`}>
                        {dataItem.anc_id}
                    </Link>
                </td>
                <td className="default-width">{getNumberOfDaysSinceDate(dataItem.EventDate)}</td>
                <td className="default-width">{dataItem.health_worker_location_name}</td>
                <td className="default-width">{dataItem.mother_symptoms}</td>
                <td className="default-width">{dataItem.health_worker_location_name}</td>
                <td className="default-width">
                    <RiskColoring {...{ risk: dataItem.logface_risk }} />
                </td>
            </tr>
        );
    };

    /**
     * Returns a <tr></tr> for an SmsData object passed to it
     * which is used to build the table above.
     * for the NBC & PNC_CHILD module
     * @param {SmsData} dataItem - an SmsData object used to generate a table row
     * @return {JSX.Element} table row
     */
    public nbcAndPncChildMapFunction = (dataItem: SmsData): JSX.Element => {
        return (
            <tr key={dataItem.event_id}>
                <td className="default-width">
                    <Link to={`${getModuleLink(this.props.module)}/child_patient_detail/${dataItem.anc_id}`}>
                        {dataItem.anc_id}
                    </Link>
                </td>
                <td className="default-width">{getNumberOfDaysSinceDate(dataItem.EventDate)}</td>
                <td className="default-width">{dataItem.health_worker_location_name}</td>
                <td className="default-width">{dataItem.child_symptoms}</td>
                <td className="default-width">{dataItem.health_worker_location_name}</td>
                <td className="default-width">
                    <RiskColoring {...{ risk: dataItem.logface_risk }} />
                </td>
            </tr>
        );
    };

    /**
     * Returns a <tr></tr> for an SmsData object passed to it
     * which is used to build the table above.
     * for the PREGNANCY module
     * @param {SmsData} dataItem - an SmsData object used to generate a table row
     * @return {JSX.Element} table row
     */
    public pregnancyMapFunction = (dataItem: SmsData): JSX.Element => {
        return (
            <tr key={dataItem.event_id}>
                <td className="default-width">
                    <Link to={`${getModuleLink(this.props.module)}/patient_detail/${dataItem.anc_id}`}>
                        {dataItem.anc_id}
                    </Link>
                </td>
                <td className="default-width">{dataItem.gravidity}</td>
                <td className="default-width">{dataItem.parity}</td>
                <td className="default-width">{dataItem.health_worker_location_name}</td>
                <td className="default-width">{dataItem.lmp_edd}</td>
                <td className="default-width">{dataItem.previous_risks}</td>
                <td className="default-width">
                    <RiskColoring {...{ risk: dataItem.logface_risk }} />
                </td>
            </tr>
        );
    };

    /**
     * Returns a <tr></tr> for an SmsData object passed to it
     * which is used to build the table above.
     * for the NUTRITION module
     * @param {SmsData} dataItem - an SmsData object used to generate a table row
     * @return {JSX.Element} table row
     */
    public nutritionMapFunction = (dataItem: SmsData): JSX.Element => {
        return (
            <tr key={dataItem.event_id}>
                <td className="default-width">
                    <Link to={`${getModuleLink(this.props.module)}/child_patient_detail/${dataItem.anc_id}`}>
                        {dataItem.anc_id}
                    </Link>
                </td>
                <td className="default-width">{getNumberOfDaysSinceDate(dataItem.EventDate)}</td>
                <td className="default-width">{dataItem.health_worker_location_name}</td>
                <td className="default-width">{dataItem.weight}</td>
                <td className="default-width">{dataItem.health_worker_location_name}</td>
                <td className="default-width">
                    <RiskColoring {...{ risk: dataItem.logface_risk }} />
                </td>
            </tr>
        );
    };
}
