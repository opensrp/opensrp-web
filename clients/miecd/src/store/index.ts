import { connectReducer, getConnectedStore } from '@onaio/connected-reducer-registry';
import { gateKeeperReducer, gateKeeperReducerName } from '@onaio/gatekeeper';
import reducerRegistry, { combine, Registry } from '@onaio/redux-reducer-registry';
import session, { reducerName as sessionReducer } from '@onaio/session-reducer';

/** Initial reducers in the reducer registry */
const defaultReducers: Registry = {
    router: connectReducer as any /** Dirty hack because connectRouter LocationChangeAction does not extend Redux.AnyAction */,
};

/** Add users reducer to registry */
defaultReducers[sessionReducer] = session;
defaultReducers[gateKeeperReducerName] = gateKeeperReducer;

/** The initial store for the reveal web app */
const store = getConnectedStore(defaultReducers);

/** Set listener to add reducers to store when registered */
reducerRegistry.setChangeListener(reducers => {
    store.replaceReducer(combine(reducers));
});

export default store;
