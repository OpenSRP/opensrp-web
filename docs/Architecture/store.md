# overview

This documentation will take you through the process of creating and working with:

    - reducers
    - actions
    - selectors

## Where do I find the store

the store reducer modules are found within src/store. The initial directory structure looks something like this: (Note: this structure only depicts the initial directory structure and will as expected contain more directories and files as more are added with time)

```
...
├── src
│   ├── ...
│   ├── store
│   │   ├── ducks
│   │   │    ├── superset.ts
│   │   │    ├── tests/
│   │   ├── tests
│   │   ├── index.ts
│   │   └── selectors.ts
...
```

A single reducer module will contain the afore mentioned parts i.e. its respective actions, action creators if any, its selectors and the reducer function

## Actions

An action can be any object with a type property, they are the only way provisioned for changing the state in a redux store.

### Adding Actions

Its a common convention as well as recommended one to Use type constants, [read more on this here]().

steps:

- Add An interface describing the action structure: redux exposes an `AnyAction` type that can be extended to write custom action interfaces
- Create a general actiontypes to be used to type check all actions that will be passed to the reducer function as an argument. this will be in the form of [intersection type]()

## Selectors

whereas the action puts data in the store; the selectors on the other hand act as getters of the slice of data that this reducer module handles

selectors are functions that will take at least one argument, the store, Then based on any other arguments like say an id, can be used to filter down the data retrieved.

## The Reducer

The reducer function, takes 2 arguments; the state and an action.
using the type property of the action, the reducer should then know how to modify the state.
such functionality could be implemented as a form of switch strcture such as a list of if statements or the more recommended approach which is to use the actual switch statement

### Adding the Reducer

- Define an interface for the data to be stored in the store
- Define an interface for the store state.
- write the actual reducer function, while applying the required type annotations as required.
