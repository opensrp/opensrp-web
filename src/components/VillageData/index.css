#navrow {
  display: flex;
  justify-content: center;
}

.village .card {
  width: 100%;
}

.villageDataRow {
  margin-top: 3% !important;
}

.village td:nth-of-type(2),
.village th:nth-of-type(2),
.village td:nth-of-type(3),
.village th:nth-of-type(3),
.village td:nth-of-type(7),
.village th:nth-of-type(7),
.village td:nth-of-type(5),
.village th:nth-of-type(5) {
  width: 6%;
}

.village td:first-of-type,
.village th:first-of-type {
  width: 9%;
}
