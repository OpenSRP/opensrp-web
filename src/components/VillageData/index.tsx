import React from 'react';
import { Link } from 'react-router-dom';
import { Card, CardBody, CardTitle, Row, Table } from 'reactstrap';
import {
  EDD,
  GRAVIDITY,
  LOCATION,
  NBC_AND_PNC_CHILD,
  PARITY,
  PATIENT_ID,
  PREGNANCY,
  PREVIOUS_PREGNANCY_RISK,
  RISK_CATEGORY,
} from '../../constants';
import { SmsData } from '../../store/ducks/sms_events';
import { getModuleLink } from '../DataCircleCard';
import { PaginationData, Paginator, PaginatorProps } from '../Paginator';
import RiskColoring from '../RiskColoring';
import './index.css';

interface Props {
  current_level: number;
  smsData: SmsData[];
  module: string;
}

interface State {
  currentPage: number;
}

const defaultProps: Props = {
  current_level: 0,
  module: '',
  smsData: [],
};

export default class VillageData extends React.Component<Props, State> {
  public static defaultProps = defaultProps;

  constructor(props: Props) {
    super(props);
    this.state = {
      currentPage: 1,
    };
  }

  public render() {
    const routePaginatorProps: PaginatorProps = {
      endLabel: 'last',
      nextLabel: 'next',
      onPageChange: (paginationData: PaginationData) => {
        this.setState({
          currentPage: paginationData.currentPage,
        });
      },
      pageLimit: 5,
      pageNeighbours: 3,
      previousLabel: 'previous',
      startLabel: 'first',
      totalRecords: this.props.smsData.length,
    };
    return (
      <React.Fragment>
        {this.props.current_level === 3 ? (
          <React.Fragment>
            <Row className="village villageDataRow">
              <Card className="table-card">
                <CardTitle>Selected Commune</CardTitle>
                <CardBody>
                  <Table striped={true} borderless={true}>
                    <thead id="header">
                      {this.props.module === PREGNANCY ? (
                        <tr>
                          <th className="default-width">{PATIENT_ID}</th>
                          <th className="default-width">{GRAVIDITY}</th>
                          <th className="default-width">{PARITY}</th>
                          <th className="default-width">{LOCATION}</th>
                          <th className="default-width">{EDD}</th>
                          <th className="default-width">{PREVIOUS_PREGNANCY_RISK}</th>
                          <th className="default-width">{RISK_CATEGORY}</th>
                        </tr>
                      ) : (
                        <tr>
                          <th className="default-width">{PATIENT_ID}</th>
                          <th className="default-width">{'Days since birth'}</th>
                          <th className="default-width">{'Location of residence'}</th>
                          <th className="default-width">{'current symptoms'}</th>
                          <th className="default-width">{'Location of birth'}</th>
                          <th className="default-width">{RISK_CATEGORY}</th>
                        </tr>
                      )}
                    </thead>
                    <tbody id="body">
                      {this.props.smsData.length
                        ? this.props.smsData
                            .slice(
                              (this.state.currentPage - 1) * routePaginatorProps.pageLimit,
                              (this.state.currentPage - 1) * routePaginatorProps.pageLimit +
                                routePaginatorProps.pageLimit
                            )
                            .map(
                              this.props.module === PREGNANCY
                                ? this.pregnancyMapFunction
                                : this.props.module === NBC_AND_PNC_CHILD
                                ? this.nbcAndPncChildMapFunction
                                : this.nbcAndPncMotherMapFunction
                            )
                        : null}
                    </tbody>
                  </Table>
                </CardBody>
              </Card>
            </Row>
            <Row id="navrow" className="villageDataRow">
              <Paginator {...routePaginatorProps} />
            </Row>
          </React.Fragment>
        ) : null}
      </React.Fragment>
    );
  }

  private nbcAndPncMotherMapFunction = (dataItem: SmsData) => {
    return (
      <tr key={dataItem.event_id}>
        <td className="default-width">
          <Link to={`${getModuleLink(this.props.module)}/patient_detail/${dataItem.anc_id}`}>
            {dataItem.anc_id}
          </Link>
        </td>
        <td className="default-width">{this.getNumberOfDaysSinceDate(dataItem.EventDate)}</td>
        <td className="default-width">{dataItem.health_worker_location_name}</td>
        <td className="default-width">{dataItem.mother_symptoms}</td>
        <td className="default-width">{dataItem.health_worker_location_name}</td>
        <td className="default-width">
          <RiskColoring {...{ risk: dataItem.logface_risk }} />
        </td>
      </tr>
    );
  };

  private nbcAndPncChildMapFunction = (dataItem: SmsData) => {
    return (
      <tr key={dataItem.event_id}>
        <td className="default-width">
          <Link to={`${getModuleLink(this.props.module)}/patient_detail/${dataItem.anc_id}`}>
            {dataItem.anc_id}
          </Link>
        </td>
        <td className="default-width">{this.getNumberOfDaysSinceDate(dataItem.EventDate)}</td>
        <td className="default-width">{dataItem.health_worker_location_name}</td>
        <td className="default-width">{dataItem.child_symptoms}</td>
        <td className="default-width">{dataItem.health_worker_location_name}</td>
        <td className="default-width">
          <RiskColoring {...{ risk: dataItem.logface_risk }} />
        </td>
      </tr>
    );
  };

  private getNumberOfDaysSinceDate(date: string): number {
    return Math.floor((new Date().getTime() - new Date(date).getTime()) / (1000 * 3600 * 24));
  }

  private pregnancyMapFunction = (dataItem: SmsData) => {
    return (
      <tr key={dataItem.event_id}>
        <td className="default-width">
          <Link to={`${getModuleLink(this.props.module)}/patient_detail/${dataItem.anc_id}`}>
            {dataItem.anc_id}
          </Link>
        </td>
        <td className="default-width">{dataItem.gravidity}</td>
        <td className="default-width">{dataItem.parity}</td>
        <td className="default-width">{dataItem.health_worker_location_name}</td>
        <td className="default-width">{dataItem.lmp_edd}</td>
        <td className="default-width">{dataItem.previous_risks}</td>
        <td className="default-width">
          <RiskColoring {...{ risk: dataItem.logface_risk }} />
        </td>
      </tr>
    );
  };
}
