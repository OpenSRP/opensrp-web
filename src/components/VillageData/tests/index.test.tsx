import { mount, shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import { createBrowserHistory } from 'history';
import React from 'react';
import { Provider } from 'react-redux';
import { Router } from 'react-router';
import VillageData, { Props } from '..';
import store from '../../../store/index';
import VillageDataProps from './villageDataPropsfixtures';

const history = createBrowserHistory();

describe('components/VillageData', () => {
  beforeEach(() => {
    jest.resetAllMocks();
  });

  it('must render without crashing', () => {
    shallow(<VillageData />);
  });

  it('must render correctly with no data', () => {
    const wrapper = mount(<VillageData />);
    expect(toJson(wrapper)).toMatchSnapshot('Village data with no data');
  });

  it('must render correctly with data', () => {
    const wrapper = mount(
      <Provider store={store}>
        <Router history={history}>
          <VillageData {...(VillageDataProps as Props)} />
        </Router>
      </Provider>
    );
    expect(toJson(wrapper.find('tr'))).toMatchSnapshot('VillageData with data');
  });
});
