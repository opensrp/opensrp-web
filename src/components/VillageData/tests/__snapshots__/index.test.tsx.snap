// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/VillageData must render correctly with data: VillageData with data 1`] = `
Array [
  <tr>
    <th
      className="default-width"
    >
      Patient ID
    </th>
    <th
      className="default-width"
    >
      Gravidity
    </th>
    <th
      className="default-width"
    >
      Parity
    </th>
    <th
      className="default-width"
    >
      Location
    </th>
    <th
      className="default-width"
    >
      EDD
    </th>
    <th
      className="default-width"
    >
      Previous Pregnancy Risk
    </th>
    <th
      className="default-width"
    >
      Risk category
    </th>
  </tr>,
  <tr
    key="1573740273076"
  >
    <td
      className="default-width"
    >
      <Link
        to="/pregnancy_compartments/patient_detail/100RH2"
      >
        <a
          href="/pregnancy_compartments/patient_detail/100RH2"
          onClick={[Function]}
        >
          100RH2
        </a>
      </Link>
    </td>
    <td
      className="default-width"
    >
      0
    </td>
    <td
      className="default-width"
    >
      0
    </td>
    <td
      className="default-width"
    >
      Thôn 1
    </td>
    <td
      className="default-width"
    >
      0
    </td>
    <td
      className="default-width"
    >
       NULL
    </td>
    <td
      className="default-width"
    >
      <riskColoring
        risk=""
      >
        <span
          className="badge badge-info"
          id="default-width"
        />
      </riskColoring>
    </td>
  </tr>,
  <tr
    key="1573740903368"
  >
    <td
      className="default-width"
    >
      <Link
        to="/pregnancy_compartments/patient_detail/100RJ0"
      >
        <a
          href="/pregnancy_compartments/patient_detail/100RJ0"
          onClick={[Function]}
        >
          100RJ0
        </a>
      </Link>
    </td>
    <td
      className="default-width"
    >
      0
    </td>
    <td
      className="default-width"
    >
      0
    </td>
    <td
      className="default-width"
    >
      Thôn 1
    </td>
    <td
      className="default-width"
    >
      0
    </td>
    <td
      className="default-width"
    >
       NULL
    </td>
    <td
      className="default-width"
    >
      <riskColoring
        risk="no risk"
      >
        <span
          className="badge badge-success"
          id="default-width"
        >
          <div>
            <p>
              No Risk
            </p>
          </div>
        </span>
      </riskColoring>
    </td>
  </tr>,
  <tr
    key="1573741222743"
  >
    <td
      className="default-width"
    >
      <Link
        to="/pregnancy_compartments/patient_detail/100RKX"
      >
        <a
          href="/pregnancy_compartments/patient_detail/100RKX"
          onClick={[Function]}
        >
          100RKX
        </a>
      </Link>
    </td>
    <td
      className="default-width"
    >
      0
    </td>
    <td
      className="default-width"
    >
      0
    </td>
    <td
      className="default-width"
    >
      Thôn 1
    </td>
    <td
      className="default-width"
    >
      0
    </td>
    <td
      className="default-width"
    >
       NULL
    </td>
    <td
      className="default-width"
    >
      <riskColoring
        risk="low"
      >
        <span
          className="badge badge-warning"
          id="default-width"
        >
          <div>
            <p>
              Low Risk
            </p>
          </div>
        </span>
      </riskColoring>
    </td>
  </tr>,
  <tr
    key="1573742257475"
  >
    <td
      className="default-width"
    >
      <Link
        to="/pregnancy_compartments/patient_detail/100RMT"
      >
        <a
          href="/pregnancy_compartments/patient_detail/100RMT"
          onClick={[Function]}
        >
          100RMT
        </a>
      </Link>
    </td>
    <td
      className="default-width"
    >
      0
    </td>
    <td
      className="default-width"
    >
      0
    </td>
    <td
      className="default-width"
    >
      Đăk Tờ Re
    </td>
    <td
      className="default-width"
    >
      0
    </td>
    <td
      className="default-width"
    >
       NULL
    </td>
    <td
      className="default-width"
    >
      <riskColoring
        risk="low"
      >
        <span
          className="badge badge-warning"
          id="default-width"
        >
          <div>
            <p>
              Low Risk
            </p>
          </div>
        </span>
      </riskColoring>
    </td>
  </tr>,
  <tr
    key="1573742379532"
  >
    <td
      className="default-width"
    >
      <Link
        to="/pregnancy_compartments/patient_detail/100RNP"
      >
        <a
          href="/pregnancy_compartments/patient_detail/100RNP"
          onClick={[Function]}
        >
          100RNP
        </a>
      </Link>
    </td>
    <td
      className="default-width"
    >
      0
    </td>
    <td
      className="default-width"
    >
      0
    </td>
    <td
      className="default-width"
    >
      Đăk Tờ Re
    </td>
    <td
      className="default-width"
    >
      0
    </td>
    <td
      className="default-width"
    >
       NULL
    </td>
    <td
      className="default-width"
    >
      <riskColoring
        risk="Not Set"
      >
        <span
          className="badge badge-info"
          id="default-width"
        >
          <div>
            <p>
              Not Set
            </p>
          </div>
        </span>
      </riskColoring>
    </td>
  </tr>,
]
`;

exports[`components/VillageData must render correctly with no data: Village data with no data 1`] = `
<VillageData
  current_level={0}
  module=""
  smsData={Array []}
/>
`;
